### DISCUSSION TITLE
Pathloss and distance for each ray

### DISCUSSION THREAD

**Discussion by fightingsoul042** [NVlabs/sionna#380] [General]
Hi, 

I wanted to ask is it possible to get the pathloss and propagation distance for each ray. Like in the following example from matlab 
https://se.mathworks.com/help/comm/ug/indoor-mimo-ofdm-communication-link-using-ray-tracing.html

Thank you.

**Comment by merlinND** [comment]
Hello @fightingsoul042,

Does the following discussion help compute the quantities you need?  :arrow_right: https://github.com/NVlabs/sionna/discussions/360

**Reply by fightingsoul042** [reply]
@merlinND I believe it might help with calculating distance for each ray. But now sure if it could help in calculating the pathloss for each ray, i.e. the power lost due to propagation and reflection.

**Comment by jhoydis** [comment]
Hi @fightingsoul042,

Actually, `paths.a` gives you the complex gain for each ray and `paths.tau` the corresponding delay.
The pathloss in dB would then simply be `-20*np.log10(np.abs(paths.a)` and the distance `paths.tau*sionna.constants.SPEED_OF_LIGHT`.

**Reply by fightingsoul042** [reply]
Hi @jhoydis, 
Thank you for the input. If I am not mistaking the paths.a, also includes gain from the antenna array. Is it possible to separate the gains contributed by the antenna array.

**Reply by jhoydis** [reply]
No, it does not contain any array gain. Paths are computed between antenna elements, not arrays. That means, however, that the antenna pattern is included in `paths.a`. If you choose an isotropic antenna pattern, the antenna has no effect.

**Reply by fightingsoul042** [reply]
@jhoydis Thank you, this helps.

**Reply by sathibabu9** [reply]
May i use the following expression PL_RAYTRACE = -10*np.log10(np.sum(np.abs(a)**2)) for a MISO narrowband channel

**Comment by sathibabu9** [comment]
I’m using Sionna RT with scenes loaded from XML (large outdoor / UTM-scale scenes). Calling

```py
scene.preview()      
```
does not show any visualization.

I found that manually placing a camera using the scene bounding box works:
```py
bbox = scene.mi_scene.bbox()
center = 0.5 * (bbox.min + bbox.max)
from sionna.rt import Camera
cam = Camera(
    position=[center[0], center[1] - 500, center[2] + 300],
    look_at=center
)

scene.render(camera=cam)
```

This renders the scene correctly.

My questions are:

Is this the recommended / intended way to preview RT scenes when scene.preview() fails?

Is there a scene-independent utility (or best practice) for automatically positioning a camera based on the scene extent?

After computing paths with PathSolver, is it safe to reuse the same camera to preview the scene again (optionally with rays using scene.render(camera=cam, paths=paths))?

Are there known limitations of scene.preview() for large scenes, merged shapes, or XML-loaded environments?

Any guidance or official recommendations would be appreciated.

**Reply by merlinND** [reply]
Hello @sathibabu9,

Your questions do not seem related to this current discussion; please open a new one.
